cpp/src/parquet/column_reader_test.cc

```diff
@@ -572,5 +576,615 @@ TEST_F(TestPrimitiveReader, TestNonDictionaryEncodedPagesWithExposeEncoding) {
   pages_.clear();
 }
 
+// Tests reading a repeated field using the RecordReader.
+TEST(RecordReaderTest, BasicReadRepeatedField) {
+  internal::LevelInfo level_info;
+  level_info.def_level = 1;
+  level_info.rep_level = 1;
+
+  NodePtr type = schema::Int32("b", Repetition::OPTIONAL);
+  const ColumnDescriptor descr(type, level_info.def_level, level_info.rep_level);
+
+  // Records look like: {[10], [20, 20], [30, 30, 30]}
+  std::vector<std::shared_ptr<Page>> pages;
+  std::vector<int32_t> values = {10, 20, 20, 30, 30, 30};
+  std::vector<int16_t> def_levels = {1, 1, 1, 1, 1, 1};
+  std::vector<int16_t> rep_levels = {0, 0, 1, 0, 1, 1};
+
+  std::unique_ptr<PageReader> pager;
+  std::shared_ptr<DataPageV1> page = MakeDataPage<Int32Type>(
+      &descr, values, /*num_values=*/static_cast<int>(def_levels.size()), Encoding::PLAIN,
+      /*indices=*/{},
+      /*indices_size=*/0, def_levels, level_info.def_level, rep_levels,
+      level_info.rep_level);
+  pages.push_back(std::move(page));
+  pager.reset(new test::MockPageReader(pages));
+
+  std::shared_ptr<internal::RecordReader> record_reader =
+      internal::RecordReader::Make(&descr, level_info);
+  record_reader->SetPageReader(std::move(pager));
+
+  int64_t records_read = record_reader->ReadRecords(/*num_records=*/2);
+
+  ASSERT_EQ(records_read, 2);
+  ASSERT_EQ(record_reader->values_written(), 3);
+  ASSERT_EQ(record_reader->null_count(), 0);
+  ASSERT_EQ(record_reader->levels_written(), 6);
+  ASSERT_EQ(record_reader->levels_position(), 3);
+
+  const auto read_values = reinterpret_cast<const int32_t*>(record_reader->values());
+  std::vector<int32_t> read_vals(read_values,
+                                 read_values + record_reader->values_written());
+  std::vector<int16_t> read_defs(
+      record_reader->def_levels(),
+      record_reader->def_levels() + record_reader->levels_position());
+  std::vector<int16_t> read_reps(
+      record_reader->rep_levels(),
+      record_reader->rep_levels() + record_reader->levels_position());
+
+  ASSERT_THAT(read_vals, ElementsAre(10, 20, 20));
+  ASSERT_THAT(read_defs, ElementsAre(1, 1, 1));
+  ASSERT_THAT(read_reps, ElementsAre(0, 0, 1));
```

Comment:
I see this pattern is used several times in many tests.
Can you instead define a test fixture with a helper method such as:
```c++
template <typename T>
void AssertReadValues(
    const RecordReader&,
    const std::vector<T>& expected_values,
    const std::vector<int16_t>& expected_def_levels,
    const std::vector<int16_t>& expected_rep_levels);
```

Reply:
More generally, it is desirable to factor out boilerplate in tests so that tests are:
1) more readable
2) more maintainable